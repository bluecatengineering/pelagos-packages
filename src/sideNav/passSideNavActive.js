import {Children, cloneElement} from 'react';

const components = ['SideNavItems', 'SideNavMenu', 'SideNavMenuItem', 'SideNavLink'];

export default (sideNavActive, children) =>
	sideNavActive
		? Children.map(children, (child) =>
				components.includes(child.type?.displayName) ? cloneElement(child, {sideNavActive}) : child
		  )
		: children;
