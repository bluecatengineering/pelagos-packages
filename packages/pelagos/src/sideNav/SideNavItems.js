import {forwardRef} from 'react';
import PropTypes from 'prop-types';

import passSideNavActive from './passSideNavActive';
import './SideNav.less';

/** Container for a list of any mix of SideNavMenu, SideNavLink, or SideNavDivider. */
const SideNavItems = forwardRef(({className, sideNavActive, children, ...props}, ref) => (
	<ul {...props} className={`SideNav__items${className ? ` ${className}` : ''}`} ref={ref}>
		{passSideNavActive(sideNavActive, children)}
	</ul>
));

SideNavItems.displayName = 'SideNavItems';

SideNavItems.propTypes = {
	/** The component class name(s). */
	className: PropTypes.string,
	/** Whether the side navigation is active. */
	sideNavActive: PropTypes.bool,
	/** The child elements. */
	children: PropTypes.node,
};

export default SideNavItems;
