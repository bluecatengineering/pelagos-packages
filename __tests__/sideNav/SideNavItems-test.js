import {shallow} from 'enzyme';

import SideNavItems from '../../src/sideNav/SideNavItems';
import passSideNavActive from '../../src/sideNav/passSideNavActive';

jest.unmock('../../src/sideNav/SideNavItems');

passSideNavActive.mockReturnValue('passSideNavActive');

describe('SideNavItems', () => {
	describe('rendering', () => {
		it('renders expected elements', () => {
			const wrapper = shallow(<SideNavItems sideNavActive>child</SideNavItems>);
			expect(wrapper.getElement()).toMatchSnapshot();
			expect(passSideNavActive.mock.calls).toEqual([[true, 'child']]);
		});

		it('renders expected elements when className is set', () => {
			const wrapper = shallow(<SideNavItems className="TestClass">child</SideNavItems>);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});
});
