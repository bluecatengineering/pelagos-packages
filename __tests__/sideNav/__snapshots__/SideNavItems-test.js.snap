// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavItems rendering renders expected elements 1`] = `
<ul
  className="SideNav__items"
>
  passSideNavActive
</ul>
`;

exports[`SideNavItems rendering renders expected elements when className is set 1`] = `
<ul
  className="SideNav__items TestClass"
>
  passSideNavActive
</ul>
`;
